keps/sig-auth/20190730-oidc-discovery.md

```diff
@@ -138,16 +175,31 @@ In addition, the API server will serve a JWKS consisting of the public keys from
 public keys used for valid KSA tokens. The OIDC discovery document will point to
 the JWKS path as the `jwks_uri`.
 
+Finally, it will be possible for users to override the `jwks_uri` via a new
+`--service-account-jwks-uri` flag. We do attempt to construct a default JWKS URI
+from the API server's external address, but without this additional option, it
+isn't possible to correctly configure the server in some cases. For example,
+even given an issuer URL, we don't necessarily know the JWKS path, because it is
+not part of the specification and the issuer URL may not point directly to an
+API server (the user may choose an alternative issuer URL if their cluster does
+not serve on the public internet, as long as the cluster is configured to use
+that URL for issuer claims).
+
 For example, if the API server is configured with the `--service-account-issuer`
-value `https://dev.cluster.internal`, the API server could expose the following
-configuration:
+value `https://cluster.example.com`, which is also the API server root, and
+`--service-account-jwks-uri` value `https://cluster.example.com/openid/v1/jwks`
+(the cluster serves the JWKS at the `/openid/v1/jwks` path by default), the
```

Comment:
How did we arrive at `/openid/v1/jwks`?

Reply:
* /openid to group any future endpoints related to this
* /v1 to version it so if format or structure or spec changes, we can publish a second version (also makes it clearer where this falls under the deprecation policy)
* /jwks to identify particular content

Reply:
SGTM